zerver/lib/users.py

```diff
@@ -399,11 +394,6 @@ def bulk_access_users_by_id(
         "realm",
         "realm__can_access_all_users_group",
         "realm__can_access_all_users_group__named_user_group",
-        "realm__direct_message_initiator_group",
-        "realm__direct_message_initiator_group__named_user_group",
-        "realm__direct_message_permission_group",
-        "realm__direct_message_permission_group__named_user_group",
-        "bot_owner",
```

Comment:
Probably these functions could benefit from using a `bulk_access_user_queryset` function exported by `zerver/models/users.py`, so this can be defined next to the other one, with comments explaining the difference. (Which is that the bulk ones are intended for a user other than the acting user).

Reply:
Like this : 
```python
def bulk_access_user_queryset():
    return UserProfile.objects.select_related(
        "realm",
        "realm__can_access_all_users_group",
        "realm__can_access_all_users_group__named_user_group",
    )
```

I don't really see a need to make a separate function just because only 2 places use the same (almost) filter

Reply:
It's because it's good to have consistent abstractions.